overcooked_ai_py/mdp/overcooked_mdp.py

```diff
@@ -669,27 +696,151 @@ def get_state_transition(self, state, joint_action):
+        - Onion is needed (all pots are non-full)
+        - Nobody is holding onions
+        """
+        if self.num_players != 2: return False
+        all_non_full = len(self.get_full_pots(pot_states)) == 0
+        other_player = state.players[1 - player_index]
+        other_player_holding_onion = other_player.has_object() and other_player.get_object().name == "onion"
+        return all_non_full and not other_player_holding_onion
+
+    def is_onion_pickup_useful(self, state, pot_states, player_index):
+        """
+        NOTE: this only works if self.num_players == 2
+        Always useful unless:
+        - All pots are full & other agent is not holding a dish
+        """
+        if self.num_players != 2: return False
+        all_pots_full = self.num_pots == len(self.get_full_pots(pot_states))
+        other_player = state.players[1 - player_index]
+        other_player_has_dish = other_player.has_object() and other_player.get_object().name == "dish"
+        if all_pots_full and not other_player_has_dish:
+            return False
+        return True
+        
+    def is_onion_drop_useful(self, state, pot_states, player_index):
+        """
+        NOTE: this only works if self.num_players == 2
+        Useful if:
+        - Dish is needed (all pots are full)
+        - Nobody is holding a dish
+        """
+        if self.num_players != 2: return False
+        all_pots_full = len(self.get_full_pots(pot_states)) == self.num_pots
+        other_player = state.players[1 - player_index]
+        other_player_holding_dish = other_player.has_object() and other_player.get_object().name == "dish"
+        return all_pots_full and not other_player_holding_dish
+
+    def soup_ready_at_location(self, state, pos):
+        if not state.has_object(pos):
+            return False
+        obj = state.get_object(pos)
+        assert obj.name == 'soup', 'Object in pot was not soup'
+        _, num_items, cook_time = obj.state
+        return num_items == self.num_items_for_soup and cook_time >= self.soup_cooking_time
+
+    def log_object_pickup(self, events_infos, state, obj_name, pot_states, player_index):
+        """Player picked an object up from a counter or a dispenser"""
+        if obj_name == "soup":
+            events_infos["soup_pickup"][player_index] = True
+
+        elif obj_name == "dish":
+            events_infos["dish_pickup"][player_index] = True
+            if self.is_dish_pickup_useful(state, pot_states):
+                events_infos["useful_d_pickup"][player_index] = True
+        
+        elif obj_name == "onion":
+            events_infos["onion_pickup"][player_index] = True
+            if self.is_onion_pickup_useful(state, pot_states, player_index):
+                events_infos["useful_o_pickup"][player_index] = True
+
+        elif obj_name == "tomato":
+            # TODO: logging/events for tomatoes are not currently supported
+            pass
+        
+        else:
+            raise ValueError()
```

Comment:
```suggestion
        obj_pickup_key = obj_name + "_pickup"
        if obj_pickup_key not in events_infos:
            raise ValueError("Unknown event {}".format(obj_pickup_key))
        events_infos[obj_pickup_key][player_index] = True
        
        USEFUL_PICKUP_FNS = {
            "onion": self.is_onion_pickup_useful,
            "dish": self.is_dish_pickup_useful
        }
        if obj_name in USEFUL_PICKUP_FNS:
            if USEFUL_PICKUP_FNS[obj_name](state, pot_states, player_index):
                obj_useful_key = "useful_" + obj_name + "_pickup"
                events_infos[obj_useful_key][player_index] = True
```

Note this requires you to change the key names from `useful_o_pickup` to `useful_onion_pickup` (and same for dishes), and also requires you to add an unused `player_index` argument to `is_dish_pickup_useful`.